dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo

```diff
@@ -0,0 +1,117 @@
+within Dynawo.Electrical.Controls.PLL;
+
+model PLL "Phase-Locked Loop"
+
+  import Modelica.Blocks;
+  import Modelica.ComplexBlocks;
+  import Modelica.Constants;
+    
+  import Dynawo.Connectors;
+  import Dynawo.Types;
+
+  parameter Types.PerUnit Kp "PLL voltage calculation proportional gain";
+  parameter Types.PerUnit Ki "PLL voltage calculation integrator gain";
+  parameter Types.PerUnit OmegaMinPu "Lower frequency limit (only positive values!)";
+  parameter Types.PerUnit OmegaMaxPu "Upper frequency limit";
+  parameter Types.PerUnit OmegaRefPu "Frequency refence value";
+
+  // Inputs:
+  ComplexBlocks.Interfaces.ComplexInput uPu(re(start = u0Pu.re), im(start = u0Pu.im)) "Complex voltage at PCC (pu base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-116, 29}, extent = {{-17, -17}, {17, 17}}, rotation = 0), iconTransformation(origin = {-101, 51}, extent = {{-9, -9}, {9, 9}}, rotation = 0)));
+  Blocks.Interfaces.RealInput omegaRefPu(start = Omega0Pu) "Reference frequency of the system. Either connected to the reference machine or the center of inertia frequency or set be constant 1." annotation(
+    Placement(visible = true, transformation(origin = {-117, -70}, extent = {{-18, -18}, {18, 18}}, rotation = 0), iconTransformation(origin = {-101, -47}, extent = {{-9, -9}, {9, 9}}, rotation = 0)));
+  
+  // Outputs:
+  Blocks.Interfaces.RealOutput omegaPLLPu "Measured Frequency, PLL (pu)" annotation(
+    Placement(visible = true, transformation(origin = {160, -63}, extent = {{-17, -17}, {17, 17}}, rotation = 0), iconTransformation(origin = {106, 50}, extent = {{-8, -8}, {8, 8}}, rotation = 0)));
+  Blocks.Interfaces.RealOutput sinphi "sin(phi) aligned with terminal voltage phasor" annotation(
```

Comment:
Do you need sinphi and cosphi as outputs? Or could they just be variables?

Reply:
Sinphi and cosphi are not needed as outputs for only measuring the frequency.
But for dq-Transformation sinphi and cosphi are common input signals and in e.g. PowerFactory these signals are directly given from the PLL. Dq-transformation will be needed for the voltage source interface. Therefore, I put these signals as outputs of the PLL as it is in PowerFactory.